docs/tutorials/sparse/train_gluon.md

```diff
@@ -0,0 +1,469 @@
+
+
+```python
+class FullyConnectedSparse(mx.gluon.HybridBlock):
+    def __init__(self, in_units, units, weight_grad_stype='default'):
+        super(FullyConnectedSparse, self).__init__()
+        with self.name_scope():
+            self._units = units
+            self.weight = self.params.get('weight', shape=(in_units, units),
+                                          init=None, allow_deferred_init=True,
+                                          dtype='float32', stype='default', grad_stype=weight_grad_stype)
+            self.bias = self.params.get('bias', shape=(units),
+                                          init='zeros', allow_deferred_init=True,
+                                          dtype='float32', stype='default', grad_stype='default')
+
+    def hybrid_forward(self, F, x, weight, bias):
+        return F.sparse.dot(x, weight) + bias
+```
+
+Once again, we're using a dense `weight`, so both `FullyConnected` and `FullyConnectedSparse` will return dense array outputs. When constructing a multi-layer network therefore, only the first layer needs to be optimized for sparse inputs. Our first layer is often responsible for reducing the feature dimension dramatically (e.g. 1,000,000 features down to 128 features). We'll set the number of units in our 3 layers to be 128, 8 and 1.
+
+We will use [`timeit`](https://docs.python.org/2/library/timeit.html) to check the performance of these two variants, and analyse some [MXNet Profiler](https://mxnet.incubator.apache.org/versions/master/tutorials/python/profiler.html) traces that have been created from these benchmarks. Additionally, we will inspect the memory usage of the weights (and gradients) using the `print_memory_allocation` function defined below:
+
+
+```python
+def print_memory_allocation(net, block_idxs):
+    blocks = [net[block_idx] for block_idx in block_idxs]
+    weight_nbytes = [get_nbytes(b.weight.data()) for b in blocks]
+    weight_nbytes_pct = [b/sum(weight_nbytes) for b in weight_nbytes]
+    weight_grad_nbytes = [get_nbytes(b.weight.grad()) for b in blocks]
+    weight_grad_nbytes_pct = [b/sum(weight_grad_nbytes) for b in weight_grad_nbytes]
+    print("Memory Allocation for Weight:")
+    for i in range(len(block_idxs)):
+        print('{:7.3f} MBs ({:7.3f}%) for {:<40}'.format(weight_nbytes[i]/1000000,
+                                                              weight_nbytes_pct[i]*100,
+                                                              blocks[i].name))
+    print("Memory Allocation for Weight Gradient:")
+    for i in range(len(block_idxs)):
+        print('{:7.3f} MBs ({:7.3f}%) for {:<40}'.format(weight_grad_nbytes[i]/1000000,
+                                                              weight_grad_nbytes_pct[i]*100,
+                                                              blocks[i].name))
+```
+
+### Benchmark: `FullyConnected`
+
+
```

Comment:
Introduce what's going on in the following code block? Or does the previous go with?

Reply:
Added a short intro.